src/ui/views/Send/SendToCadence/TransferConfirmation.tsx

```diff
@@ -72,150 +76,158 @@ const TransferConfirmation = (props: TransferConfirmationProps) => {
   }, [props.data.contact.address, setCount]);
 
   const getPending = useCallback(async () => {
-    const pending = await wallet.getPendingTx();
+    const pending = await usewallet.getPendingTx();
     if (pending.length > 0) {
       setOccupied(true);
     }
-  }, [wallet]);
+  }, [usewallet]);
 
   const updateOccupied = useCallback(() => {
     setOccupied(false);
   }, []);
 
-  const transferToken = async () => {
-    // TODO: Replace it with real data
-    if (props.data.childType === 'evm') {
-      withDrawEvm();
-      return;
-    } else if (props.data.childType) {
-      sendFromChild();
-      return;
+  const transferTokenOnCadence = useCallback(async () => {
+    const amount = new BN(props.data.amount).decimalPlaces(8, BN.ROUND_DOWN).toString();
```

Comment:
This is where we have issues... You've got to change this. The amount should be rounded well before you get to this point. If anything throw an error if the rounded down amount !== the amount. 
```suggestion
    if (amount !== props.data.amount) {
      throw new Error("Invalid amount entered by the user")
    }
```